### ISSUE TITLE
Add zfs kernel module signing to enable SecureBoot mode

### ISSUE THREAD

**Issue by bsherman** [ublue-os/ucore#16]
Currently the zfs kernel modules are NOT signed, thus this image cannot both run in SecureBoot mode AND use ZFS.

Would be helpful if this were possible.

**Comment by bsherman** [comment]
This is probably not too crazy, just needs some time scripting: 

Here's a useful looking snippet: https://gist.github.com/jdoss/495e814ef4ef599e480c31d4843eb6c0

**Comment by bsherman** [comment]
Follow up on my last comment... the trick is that the openzfs provided Makefile is building the RPM... so we don't have an obvious place to inject a signing script. 

I suspect that the `./configure` options or `Makefile` provides a way to sign, but I will have to dig to find it.

**Comment by bsherman** [comment]
Follow up on my last comment... the trick is that the openzfs provided Makefile is building the RPM... so we don't have an obvious place to inject a signing script. 

I had suspected that the `./configure` options or `Makefile` provides a way to sign, so have been slowly tinkering with this.

It turns out, one can place the signing keys in the correct place in the linux kernel source tree, and signing does happen "automatically". See: https://www.kernel.org/doc/html/v4.15/admin-guide/module-signing.html

I've verified that the signing does occur, but STILL the resulting kmod rpm contains kmods which are not signed.

I'm currently digging into the idea that the RPM build process is stripping the signatures:
